actions/installer.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# SLSA verifier install script
+
+VERIFIER_RELEASE="$1"
+INSTALL_DIR="$2"
+SUDO=
+if [[ "$3" == "true" ]]; then
+  SUDO=sudo
+fi
+
+shopt -s expand_aliases
+if [ -z "$NO_COLOR" ]; then
+  alias log_info="echo -e \"\033[1;32mINFO\033[0m:\""
+  alias log_error="echo -e \"\033[1;31mERROR\033[0m:\""
+else
+  alias log_info="echo \"INFO:\""
+  alias log_error="echo \"ERROR:\""
+fi
+set -e
+
+${SUDO} mkdir -p ${INSTALL_DIR}
+cd ${INSTALL_DIR}
+
+if [[ "${VERIFIER_RELEASE}" == "main" ]]; then
+    log_info "installing verifier via 'go install' from its main version"
+    GOBIN=$(go env GOPATH)/bin
+    go install github.com/slsa-framework/slsa-verifier/cli/slsa-verifier@main
+    ${SUDO} ln -s ${GOBIN}/slsa-verifier ${INSTALL_DIR}/slsa-verifier
+    echo "Installed SLSA verifier at ${INSTALL_DIR}/slsa-verifier"
+    exit 0
+fi
+
+# Download v1.2.0 for bootstrapping
```

Comment:
The simplest way to go about installation and reduce maintenance for us to the minimum is something like the following:
1. Download a pre-defined version of the verifier and verify its hash (may be able to call   https://github.com/slsa-framework/slsa-github-generator/tree/main/.github/actions/compute-sha256 to calculate the hash without fiddling with sha256sum yourself). This need not be updated regularly, or at all (unless a vulnerability is reported)
2. Use this version to download the version the user pinned the action to (`@v1.2.3`): we can use `GITHUB_REF_NAME`
3. Figure a way to map a hash to a release (in case of `@<hash>`), and download like in previous step. Take a look at the GitHub context to see if the tag is accessible. If not, let's look into GitHub APIs. 

@ianlewis does the above sound resoanable?

Reply:
Is `GITHUB_REF_NAME` the ref of the action or the ref from the calling workflow's context? I think it's the latter. At least for reusable workflows we had to get this ref via the OIDC token IIRC.
https://github.com/slsa-framework/slsa-github-generator/blob/c565ad10f0c2efdc6fcc8fe55b52845f85014dfa/.github/actions/detect-workflow/main.go#L104-L129

Reply:
I think you're right, it's the caller's ref. Any idea how to get our own ref without resorting to OIDC?

Reply:
Copying from another comment:

The GitHub context contains an `action_path` that looks like this:

`"action_path": "/home/runner/work/_actions/laurentsimon/hello-github-actions/v1.0.0/action-a"`, corresponding to the code below in the calling workflow. 
```
uses: laurentsimon/hello-github-actions/action-a@v1.0.0
```